packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -10,37 +10,160 @@ import 'package:flutter/material.dart' hide TextStyle;
 import 'package:intl/intl.dart';
+import 'package:provider/provider.dart';
 
+import '../../flutter/auto_dispose_mixin.dart';
+
+import 'memory_controller.dart';
 import 'memory_graph_model.dart';
 
-class FlameChart extends StatelessWidget {
+class HeatMapSizeAnalyzer extends SingleChildRenderObjectWidget {
+  const HeatMapSizeAnalyzer({
+    Key key,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  @override
+  RenderFlameChart createRenderObject(BuildContext context) {
+    return RenderFlameChart();
+  }
+}
+
+class RenderFlameChart extends RenderProxyBox {
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    if (child != null) {
+      context.paintChild(child, offset);
+    }
+  }
+}
+
+class FlameChart extends StatefulWidget {
   const FlameChart(
-    this.sizes, {
-    // Flame chart has a blueish color.
-    this.lightColor = const Color(0xFFBBDEFB),
-    this.darkColor = const Color(0xFF0D47A1),
-  });
+    this.controller,
+  );
 
-  final InstructionsSize sizes;
-  final Color lightColor;
-  final Color darkColor;
+  final MemoryController controller;
+
+  @override
+  FlameChartState createState() => FlameChartState(controller);
+}
+
+class FlameChartState extends State<FlameChart> with AutoDisposeMixin {
+  FlameChartState(this.controller);
+
+  InstructionsSize sizes;
+
+  Map<String, Function> callbacks = {};
+
+  /// Flame chart has a blueish color.
+  Color lightColor = const Color(0xFFBBDEFB);
+
+  Color darkColor = const Color(0xFF0D47A1);
+
+  MemoryController controller;
+
+  Widget snapshotDisplay;
+
+  _FlameChart _flameChart;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    controller = Provider.of<MemoryController>(context);
+
+    cancel();
+
+    addAutoDisposeListener(controller.selectedSnapshotNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(true, true);
+
+        sizes = InstructionsSize.fromSnapshop(controller);
+      });
+    });
+
+    addAutoDisposeListener(controller.filterNotifier, () {
+      setState(() {
+        controller.computeAllLibraries(true, true);
+      });
+    });
+
+    addAutoDisposeListener(controller.searchNotifier, () {
+      setState(() {
+        final searchingValue = controller.search;
+        if (searchingValue.isNotEmpty) {
+          final Node node = findNode(searchingValue);
+          if (node != null) {
+            selectNode(node);
+          }
+        }
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    return _FlameChart(sizes, lightColor, darkColor);
+    sizes = InstructionsSize.fromSnapshop(controller);
+
+    if (sizes != null) {
+      _flameChart = _FlameChart(sizes, lightColor, darkColor, callbacks);
+      return _flameChart;
+    } else {
+      return const Text('');
```

Comment:
as elsewhere prefer SizedBox to a zero character string for all cases where we are showing empty content where the non-empty content is not a string.

Reply:
Fixed it everywhere.